design/pub-sub.md

```
@@ -51,113 +51,137 @@ The process of the event pub-sub is as follows:
+	if subs, ok := m.subscriptionsMap.Get(docKey); ok {
+		subs.Publish(event)
 	}
+	
+	// ...
+}
+
+// server/backend/pubsub/publisher.go
+func (bp *BatchPublisher) Publish(event events.DocEvent) {
+        bp.mutex.Lock()
+        defer bp.mutex.Unlock()
+		
```

markdownlint-cli2 (0.17.2)

Hard tabs
Column: 1

(MD010, no-hard-tabs)